FLEDGE.md

```diff
@@ -121,6 +122,8 @@ The `biddingWasmHelperUrl` field is optional, and lets the bidder provide comput
 
 The `dailyUpdateUrl` provides a mechanism for the group's owner to periodically update the attributes of the interest group: any new values returned in this way overwrite the values previously stored (except that the `name` and `owner` cannot be changed).  However, the browser will only allow daily updates when a sufficiently large number of people have the same `dailyUpdateUrl` , e.g. at least 100 browsers with the same update URL. This will not include any metadata, so data such as the interest group `name` should be included within the URL, so long as the URL exceeds the minimum count threshold.  (Without this sort of limit, a single-person interest group could be used to observe that person's coarse-grained IP-Geo location over time.)
 
+The `executionMode` attribute is optional. The default value (`compatibility`) will run each invocation of `generateBid` in a totally fresh execution environment, which prevents them from directly passing data to each other, but has non-trivial execution costs.  The `groupByOrigin` mode will attempt to re-use the execution environment for interest groups with the same script that were added on the same site, which saves a lot of these setup costs, but in order to avoid collusion, attempts to join the same interest group in `groupByOrigin` mode from multiple sites will result in previously joined `groupByOrigin` groups being removed.
```

Comment:
In other words, setting `'executionMode': 'groupByOrigin'` means "I promise that this domain is the only domain where people get added to this Interest Group"?

Reply:
By the way, could an alternative be that in `'executionMode': 'groupByOrigin'`, interest groups for the same owner are stored namespaced additionally by the origin where a browser gets added to one? In other words, interest group `sneaker` joined on `shoes.example` and interest group `sneaker` joined on `athletics.example` would be two completely distinct interest groups.

Reply:
> 

@michaelkleber:
"I promise" is probably too strong, and makes it sound like an ethical failing of some sort to fail to follow through?  It's more like "I think it's sufficiently high probability that this interest group and all others for the same owner with the groupByOrigin execMode being added right now would not be added anywhere else that reverting all those joins if that does happen is worth the resource savings".  I suppose I should say something like that to, well, explain, why the knob exists?

@sbelov: This is messing with what the primary key for main APIs so I it's hard for me to think of all the implications.  I am a little scared of possibility of there being more interest groups, at any rate.

Reply:
I don't feel strongly either way.

Reply:
Added a note: re:intended usage